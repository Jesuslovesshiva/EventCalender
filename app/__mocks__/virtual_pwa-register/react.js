import {useState} from 'react'

const useRegisterSW = jest.fn(() => {
  const [needRefresh, setNeedRefresh] = useState(true)

  return {
    needRefresh: [needRefresh, setNeedRefresh],
    updateServiceWorker: () => {}
  }
})

// eslint-disable-next-line import/prefer-default-export
export {useRegisterSW}
