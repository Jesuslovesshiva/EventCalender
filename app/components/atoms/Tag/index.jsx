import classNames from 'classnames'
import noop from 'lodash/noop'
import PropTypes from 'prop-types'

const TAG_COLORS = {
  BLUE: 'blue',
  EMERALD: 'emerald',
  GRAY: 'gray',
  GREEN: 'green',
  ORANGE: 'orange',
  RED: 'red',
  YELLOW: 'yellow'
}
const AVAILABLE_TAG_COLORS = Object.values(TAG_COLORS)

const TAG_COLORS_CLASSES = {
  [TAG_COLORS.GRAY]: 'bg-gray-light border-gray-medium',
  [TAG_COLORS.RED]: 'bg-red-light border-red-medium',
  [TAG_COLORS.ORANGE]: 'bg-orange-light border-orange-medium',
  [TAG_COLORS.YELLOW]: 'bg-yellow-light border-yellow-medium',
  [TAG_COLORS.BLUE]: 'bg-blue-light border-blue-medium',
  [TAG_COLORS.GREEN]: 'bg-green-light border-green-medium',
  [TAG_COLORS.EMERALD]: 'bg-emerald-light border-emerald-medium'
}

const Tag = ({children, ...props}) => (
  <div
    className={classNames(
      'border-[1px] rounded p-1.5 pt-0.5 pb-px',
      'inline-flex items-center gap-0.5',
      'font-badge-text',
      TAG_COLORS_CLASSES[props.color],
      props.className
    )}
    onClick={props.onClick}
    title={props.title}
  >
    {children}
  </div>
)

Tag.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.any,
  color: PropTypes.oneOf(AVAILABLE_TAG_COLORS),
  onClick: PropTypes.func,
  title: PropTypes.string
}

Tag.defaultProps = {
  className: null,
  color: TAG_COLORS.GRAY,
  onClick: noop,
  title: null
}

export {AVAILABLE_TAG_COLORS, TAG_COLORS}
export default Tag
