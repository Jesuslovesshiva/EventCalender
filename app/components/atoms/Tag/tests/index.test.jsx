import {render} from '@testing-library/react'

import Tag from '../index'

const text = 'Tag Text'
const renderComponent = () => render(<Tag>{text}</Tag>)

describe('Tag', () => {
  it('should match the snapshot', () => {
    const {container} = renderComponent()
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render its children', () => {
    const {getByText} = renderComponent()
    expect(getByText(text)).toBeInTheDocument()
  })
})
