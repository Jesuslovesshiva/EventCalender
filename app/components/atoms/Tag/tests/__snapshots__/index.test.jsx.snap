// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tag should match the snapshot 1`] = `
<div
  class="border-[1px] rounded p-1.5 pt-0.5 pb-px inline-flex items-center gap-0.5 font-badge-text bg-gray-light border-gray-medium"
>
  Tag Text
</div>
`;
